ramp-engine/ramp_engine/base.py

```diff
@@ -61,6 +61,11 @@ def teardown(self):
         """Clean up (i.e., removing path, etc.) before killing the worker."""
         self.status = 'killed'
 
+    @abstractmethod
+    def _is_submission_terminated(self):
```

Comment:
Maybe we could call it, `_is_worker_active` or something along those lines? Because the current name makes me think that the submission was cancelled. While it was the whole machine that was terminated/cancelled as far as I understand. Or at least some other way to differentiate between this and the `worker._status`. In the context of a `DaskWorker` https://github.com/paris-saclay-cds/ramp-board/pull/452 I guess that would be something like no workers found.

Reply:
No nevermind my previous comment, I just saw there was a `_is_submission_finished` so it does look like, `_is_submission_*` functions would map to a `worker._status` directly.

Then maybe a better name would be `_is_submission_interrupted` ? Because I would have assumed that `_is_submission_terminated` would map to a `killed` status and not a `restart` one.

We would also need to add the `restart` to the list possible statuses in the BaseWorker docstring.